bigchaindb_driver/utils.py

```diff
@@ -0,0 +1,41 @@
+from bigchaindb.common.transaction import Asset
+
+
+class CreateOperation:
+    pass
+
+
+class TransferOperation:
+    pass
+
+
+ops_map = {
+    'CREATE': CreateOperation,
+    'TRANSFER': TransferOperation,
+}
+
+
+def _normalize_operation(operation):
+    try:
+        operation = operation.upper()
+    except AttributeError:
+        pass
+
+    try:
+        operation = ops_map[operation]()
+    except KeyError:
+        pass
+
+    return operation
+
+
+def _normalize_asset(asset, is_transfer=False):
+    if is_transfer:
+        asset = Asset.from_dict(asset)
+    else:
+        try:
+            asset = Asset(**asset)
+        except (AttributeError, TypeError):
+            if not asset:
+                asset = None
```

Comment:
This is ok for the scope of this pull request.
But overall there is still a bit of confusion because some parts of the code try to be isolated from common (argument validation, raising of exceptions) while other parts still rely on common to perform these tasks